lib/brakeman/app_tree.rb

```diff
@@ -96,13 +128,28 @@ def select_files(paths)
 
     def select_only_files(paths)
       return paths unless @only_files
-      paths.select { |f| @only_files.match f }
+      project_root  = Pathname.new(@root)
+      paths.select do |path|
+        absolute_path = Pathname.new(path)
+        # relative root never has a leading separator. But, we use a leading
+        # separator in a @skip_files entry to imply that a directory is
+        # "absolute" with respect to the project directory.
+        project_relative_path = File::SEPARATOR + absolute_path.relative_path_from(project_root).to_s
```

Comment:
Hmm...ok...maybe I wasn't totally crazy here :smile:. Reading my own comment above...we are forcing a leading `File::SEPERATOR` prefix since that is what we use in https://github.com/presidentbeef/brakeman/pull/764/files#diff-d300ce9778018ef3e26c917b83c13230R38 to delineate a "project relative root". And, by default, `File.join` doesn't return a leading slash. So, in order to get `File.join` to do that you have to call it using a somewhat ugly/non-obvious approach:

``` ruby
File.join("foo/bar")
=> "foo/bar"
File.join("", "foo/bar")
=> "/foo/bar"
File.join(File::SEPARATOR, "foo/bar")
=> "/foo/bar"
```

All of the approaches to force the leading slash seem pretty non-obvious (i.e. it seems to be abusing what `File.join` actually is supposed to be doing. Thoughts?